alacritty/src/input.rs

```diff
@@ -377,6 +377,20 @@ impl<'a, T: EventListener, A: ActionContext<T>> Processor<'a, T, A> {
         point.line = min(point.line, self.ctx.terminal().grid().num_lines() - 1);
 
         self.ctx.mouse_mut().click_state = match self.ctx.mouse().click_state {
+            // Don't clear selection if right mouse button is pressed
+            ClickState::Click
+                if mouse.right_button_state == ElementState::Pressed
+                    && mouse.left_button_state == ElementState::Released =>
+            {
+                if !self.ctx.modifiers().shift()
+                    && self.ctx.terminal().mode().intersects(TermMode::MOUSE_MODE)
+                {
+                    // 2 is the right button code
+                    self.mouse_report(2, ElementState::Pressed);
+                    return;
+                }
+                ClickState::Click
+            }
```

Comment:
Could you move this together with the left click logic, just excluding the statements we don't want to execute with a conditional? We don't want to duplicate code.

Reply:
Alright, I think it looks better now :)